subsys/bluetooth/host/hci_core.c

```diff
@@ -4113,7 +4113,15 @@ static int le_scan_set_random_addr(bool active_scan, uint8_t *own_addr_type)
 		    adv && !atomic_test_bit(adv->flags, BT_ADV_ENABLED)) {
 			err = le_set_private_addr(BT_ID_DEFAULT);
 			if (err) {
-				return err;
+				if (active_scan ||
+				    (adv && !atomic_test_bit(adv->flags,
```

Comment:
Sorry, another thing I noticed: the top level condition for the branch that the `le_set_private_addr()` call is in is already checking for `adv != NULL` so checking for it again seems redundant here?

Reply:
While processing this comment I realized that the above statement is wrong, there is an issue with the use of parenthesis.

I tested it with` west build -b native_posix tests/bluetooth/shell -- -DCONFIG_BT_PRIVACY=n -DCONFIG_BT_EXT_ADV=y`
Looking at the btmon there is no call te LE_Set_Random_Addr, before starting the scanner.
Added a fix, and now the NRPA is set.

I extracted it if-statement out to a helper function because repeating the same check was difficult at the indentation level.
The check should only be applied if the advertiser is using the random address, if the advertiser is using the LE_Set_Advertiser_Random_Address then this should not happen.